### ISSUE TITLE
Drawbacks facing graph federated learning in the setting of traffic speed prediction

### ISSUE THREAD

**Issue by LarryHawkingYoung** [alibaba/FederatedScope#511]
I want to run Graph Federated Learning in the setting of traffic speed prediction. In this setting we have one graph representing a  road network, each node is a road, which holds several historical speed records of this road in a consistent time period. For each road, it is a time series prediction problem(using historical series to predict future series), while for the whole graph, I'd like to integrate the spatial information(edges between nodes) with the temporal information to improve the predicting accuracy.

It seems like a traditional node-level predicting problem in graph learning, but not. In traditional ones, we use several nodes to train and other nodes to test. However, in my setting every nodes are used both to train and to test, the training set and testing set are split inside each node, which means in each node we use several historical speed records to train and others to test.

I have read the source code of LouvainSplitter in FederatedScope, its calling input is a type of torch_geometric.data.Data from pyG, and requires the input having attributes like 'train_mask', 'val_mask', 'test_mask', it is a node-style training-testing dataset splitting style in pyG, so don't you support my training-testing dataset splitting described above ? I wonder how can I use FederatedScope to  do Graph Federated Learning to solve my traffic speed time series predicting problem ?

This puzzle has confused me for so long.
Thanks for your reading and answering, please forgive me for my Chinese-style English.
 
Btw, I think LouvainSplitter just needs the edges information and node index of a global gragh, it dose not neet the nodes attributes(or features), because it returns just several subgraghs, while each node's attributes are not changed. 
So it's better not to send the whole Data into LouvainSplitter as a calling input. Then I think my problem will be solved well.
